### ISSUE TITLE
Automatic statistics spams the jobs table making it hard to see other jobs

### ISSUE THREAD

**Issue by awoods187** [cockroachdb/cockroach#34745]
**Describe the problem**

When running the master, automatic statistics causes many new jobs to be created for statistics now that it takes advantage of this mechanism. This can make `show jobs` and the webui very hard to read and miss other important jobs like schema changes.
![image](https://user-images.githubusercontent.com/22278911/52498358-9bcd4f80-2ba6-11e9-971e-028eea7e2e04.png)

https://gist.github.com/awoods187/d94b7906cf70f4ad11039a2871777155

**Environment:**
v2.2.0-alpha.20181217-1144-ge3e4f96

**Potential Solution**
I think we should only show the most recent create statistics jobs (e.g., not all 7 from table [58]) in the jobs table. Maybe it's valuable to have all this history stored elsewhere but I don't think it should be in the jobs table

**Comment by piyush-singh** [comment]
see also #34377 and #34829

**Comment by dt** [comment]
We discussed this a bit in #bulk-io since it has historically been a big user of and maintainer of the jobs infra. One long-term option is a single, long-lived job for each table's stats that toggles between states like waiting and running instead of record per run.

Shorter-term, we identified a couple fixes, most of which have been long-standing TODOs even before we added stats jobs:
1) GC old jobs records so the table doesn't grow forever. This was already an issue if someone did frequent inc backups or something on lots of tables. Done in #34725 (can tune the default if we want).
2) Modify `SHOW JOBS` to show what you're interested in -- long before stats jobs, it was already showing it's shortcomings last year on reg -- when trying to debug BACKUP jobs, we were wading though a wall of output every time and took to sticking it in a `SELECT * FROM [SHOW JOBS] WHERE <recent>`. The default, first thing you see in SHOW JOBS and the admin ui jobs page should probably be "What is happening?" / "what just happened?" since you _probably_ ran/opened since something just happened or you're waiting on something that is happening. A more historical view is probably the rarer case. #34829 starts this for `SHOW JOBS`.
3) Specific to stats jobs: add an toggle to admin ui display of eventlog and jobs pages, default checked, that says "hide automatic system-internal jobs" or something and filter those at display time.
4) specific to stats: maybe they don't belong in the event log at all? TBD.
